src/amqp.js

```diff
@@ -0,0 +1,344 @@
+   * @param {object} params.data Data object sent to queue
+   * @returns {boolean} returns true on success
+   */
+  async function sendToQueue({blobId, status, data}) {
+    debug(`sendToQueue`);
+    // eslint-disable-next-line no-useless-catch
+    try {
+      const {queue} = await generateQueueId({blobId, status});
+      debug(`Queue ${queue}`);
+      debugDev(`Queue Asserted: ${queue}`);
+
+      debugData(`Data ${JSON.stringify(data)}`);
+
+      debugDev(`Actually sendToQueue: ${queue}`);
+      await channel.sendToQueue(
+        queue,
+        Buffer.from(JSON.stringify({data})),
+        {
+          correlationId: blobId,
+          persistent: true
+        }
+      );
+      debugDev(`Send message to queue: ${queue}`);
+
+      return true;
+    } catch (error) {
+      debug(`SendToQueue errored: ${JSON.stringify(error)}`);
+      handleAmqpErrors(error);
+    }
+  }
+
+  async function removeQueue({blobId, status}, allowMessageLoss = false) {
+    const {queue, channelInfo} = await generateQueueId({blobId, status});
+    // deleteQueue borks the channel if the queue does not exist
+    // -> use throwaway tempChannel to avoid killing actual channel in use
+    // this might be doable also with assertQueue before deleteQueue
+    const tempChannel = await connection.createChannel();
+    if (!allowMessageLoss && channelContainsMessages(channelInfo)) {
+      throw new Error('Trying to remove queue that has unhandled messages!');
+    }
+
+    await tempChannel.deleteQueue(queue);
+
+    if (tempChannel) {
+      await tempChannel.close();
+      return;
+    }
+
+    return;
+  }
+
+  // ----------------
+  // Helper functions
+  // ----------------
+
+  function channelContainsMessages(channelInfo) {
+    if (channelInfo.messageCount < 1) {
+      debug(`checkQueue: ${channelInfo.messageCount} - queue is empty`);
+      return false;
+    }
+    debug(`Queue has ${channelInfo.messageCount} messages`);
+    return true;
+  }
+
+  async function generateQueueId({blobId, status}) {
+    const isUndefined = blobId === undefined || status === undefined;
+    const notString = typeof blobId !== 'string' || typeof status !== 'string';
+    const isEmpty = blobId.length < 1 || status.length < 1;
+    if (isUndefined || notString || isEmpty) {
+      throw new ApiError(httpStatus.INTERNAL_SERVER_ERROR, 'invalid operation parameters');
+    }
+    const queue = `${blobId}.${status}`;
+    const channelInfo = await channel.assertQueue(queue, {durable: true});
+    return {queue, channelInfo};
+  }
+
+  function transformMessagesToRecords(messages) {
+    debug(`Parsing messages (${messages.length}) to records`);
+
+    return messages.map(message => {
+      const content = JSON.parse(message.content.toString());
+      // Use subfieldValues: false validationOption here
+      return new MarcRecord(content.data, {subfieldValues: false});
+    });
+  }
+
+  async function getData(queue) {
```

Comment:
// MARK: Get data
  /**
   * Gets array of 100 messages from queue. CHUNK_SIZE set in constants.
   * @param {queue} Queue
   * @returns {[message]} returns chunk of messages as array
   */